csharp/extractor/Semmle.Extraction.CSharp/Entities/Compilations/CompilerDiagnostic.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections.Concurrent;
+using System.IO;
+using Semmle.Util;
+
+namespace Semmle.Extraction.CSharp.Entities
+{
+    internal class CompilerDiagnostic : FreshEntity
+    {
+        private static readonly int limit = EnvironmentVariables.TryGetExtractorOption<int>("COMPILER_DIAGNOSTIC_LIMIT") ?? 1000;
```

Comment:
Should we introduce an extractor option for this? What should be the name for the option?